airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java

```diff
@@ -125,15 +149,41 @@ private ScheduleRetrieverOutput getTimeToWaitFromScheduleType(final StandardSync
                     + MIN_CRON_INTERVAL_SECONDS
                 : currentSecondsSupplier.get()) * MS_PER_SECOND);
         final Date nextRunStart = cronExpression.getNextValidTimeAfter(new Date(earliestNextRun));
-        final Duration timeToWait = Duration.ofSeconds(
+        Duration timeToWait = Duration.ofSeconds(
             Math.max(0, nextRunStart.getTime() / MS_PER_SECOND - currentSecondsSupplier.get()));
+
+        timeToWait = addSchedulingNoiseForAllowListedWorkspace(timeToWait, standardSync);
         return new ScheduleRetrieverOutput(timeToWait);
       } catch (final ParseException e) {
         throw (DateTimeException) new DateTimeException(e.getMessage()).initCause(e);
       }
     }
   }
 
+  private Duration addSchedulingNoiseForAllowListedWorkspace(Duration timeToWait, StandardSync standardSync) {
+    final UUID workspaceId;
+    try {
+      workspaceId = workspaceHelper.getWorkspaceForConnectionId(standardSync.getConnectionId());
+    } catch (JsonValidationException | ConfigNotFoundException e) {
+      // We tolerate exceptions and fail open by doing nothing.
+      return timeToWait;
+    }
+    if (!SCHEDULING_NOISE_WORKSPACE_IDS.contains(workspaceId)) {
+      // Only apply to a specific set of workspaces.
+      return timeToWait;
+    }
+    if (!standardSync.getScheduleType().equals(ScheduleType.CRON)) {
+      // Only apply noise to cron connections.
+      return timeToWait;
+    }
+
+    // We really do want to add some scheduling noise for this connection.
+    final long minutesToWait = (long) (Math.random() * SCHEDULING_NOISE_CONSTANT);
+    LOGGER.debug("Adding {} minutes noise to wait", minutesToWait);
+    // Note: we add an extra second to make the unit tests pass in case `minutesToWait` was 0.
```

Comment:
nitpick: is this necessary?

Reply:
Without it the test fails 1/15 times (because `Math.random() * SCHEDULING_NOISE_CONSTANT` is <1, so it rounds to zero) which is pretty flaky for a unit test. It's a bit of a hack (hence the comment) - a better thing would be to inject a randomness provider or do some seeding of Math.random but this was a lot simpler.